### PR TITLE
Fix: APP-2284 - Creating DAO with wrapped token

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- Adds a fix to create a DAO with a wrapped token
- Fixes the issue of Token Voting DAOs crashing on community page

Task: [APP-2284](https://aragonassociation.atlassian.net/browse/APP-2284)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2284]: https://aragonassociation.atlassian.net/browse/APP-2284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/context/createDao.tsx [hunk 1]
```diff
@@ -215,6 +215,8 @@ const CreateDaoProvider: React.FC = ({children}) => {
   const getNewErc20PluginParams =
     useCallback((): TokenVotingPluginInstall['newToken'] => {
       const {tokenName, tokenSymbol, wallets} = getValues();
+      console.log(tokenName, wallets);
```
Comment (sepehr2github) [inline packages/web-app/src/context/createDao.tsx]: Please remove this log

File: packages/web-app/src/context/createDao.tsx [hunk 2]
```diff
@@ -231,10 +233,10 @@ const CreateDaoProvider: React.FC = ({children}) => {
     useCallback((): TokenVotingPluginInstall['useToken'] => {
       const {tokenAddress, tokenName, tokenSymbol} = getValues();
       return {
-        tokenAddress: tokenAddress.address, // contract address of the token to use as the voting token
+        tokenAddress: tokenAddress.address, // contract address of underlying token
         wrappedToken: {
-          name: tokenName, // the name of your token
-          symbol: tokenSymbol, // the symbol for your token. shouldn't be more than 5 letters
+          name: 'g' + tokenName, // the name of the wrapped token (gToken)
```
Comment (sepehr2github) [inline packages/web-app/src/context/createDao.tsx]: Selim Suggested using Governance for the token name if It's wrap and If It's not we need to use the exact name, For example ANT is not a governance token and It's a Simple ERC20 so we need to pass gATN but YTR is a governance token and full compatible so no wrapping needed and we need to pass tokenName

File: packages/web-app/src/context/createDao.tsx [hunk 3]
```diff
@@ -231,10 +233,10 @@ const CreateDaoProvider: React.FC = ({children}) => {
...
+          symbol: 'g' + tokenSymbol, // the symbol of the wrapped token (gTokenSymbol)
```
Comment (sepehr2github) [inline packages/web-app/src/context/createDao.tsx]: Same for Symbol as well

File: packages/web-app/src/context/createDao.tsx [hunk 4]
```diff
@@ -264,11 +266,14 @@ const CreateDaoProvider: React.FC = ({children}) => {
       }
       case 'token': {
         const [votingSettings, network] = getVoteSettings();
+
         const tokenVotingPlugin =
           TokenVotingClient.encoding.getPluginInstallItem(
             {
               votingSettings: votingSettings,
-              ...(tokenType === 'governance-ERC20' && !isCustomToken
+              ...((tokenType === 'governance-ERC20' || // token can be used as is
```
Comment (sepehr2github) [inline packages/web-app/src/context/createDao.tsx:283]: Thanks for the Fix

File: packages/web-app/src/hooks/useDaoMembers.tsx [hunk 5]
```diff
@@ -143,25 +144,18 @@ export const useDaoMembers = (
...
-                daoToken?.address,
-                (m as TokenVotingMember)?.address,
-                provider,
-                CHAIN_METADATA[network].nativeCurrency
-              );
-            else
-              return fetchBalance(
-                daoToken?.address,
-                m as string,
+                daoToken.address,
```
Comment (sepehr2github) [inline packages/web-app/src/hooks/useDaoMembers.tsx:157]: I don't remember if we finished this fix last night or we left it incomplete, Is the Fix finalized?

**Review by sepehr2github — CHANGES_REQUESTED** [review]
Thanks, Just some minor changes needed

File: packages/web-app/src/context/createDao.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/context/createDao.tsx]: Ah, makes sense

File: packages/web-app/src/context/createDao.tsx (hunk 4 above)
Comment (Fabricevladimir) [reply packages/web-app/src/context/createDao.tsx:283]: Wasn't that your fix anyway? lol

File: packages/web-app/src/hooks/useDaoMembers.tsx (hunk 5 above)
Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useDaoMembers.tsx:157]: Per Pieter the fix is complete

**Review by sepehr2github — APPROVED** [review]
Thanks

**Pull request merged**